modules/network/pre-existing-subnetwork/variables.tf

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+# variable "host_project_id" {
+#   description = "Project id of the host project"
+#   type        = string
+# }
+
+# variable "network_name" {
+#   description = "Name of the existing Shared VPC network"
+#   type        = string
+# }
+
+variable "subnetwork_self_link" {
+  description = "Self-link of the subnet in the Shared VPC"
+  type        = string
+  default     = null
+}
+
+variable "host_project" {
```

Comment:
I think that should be named `project`.  And it shouldn't be `optional`.

Reply:
https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/compute_subnetwork

All 4 arguments are optional - That makes host_project optional too (I can change the name).